File: vibe-lang/benchmark/output/lua-love/openai/cellular_automata.lua
local cells = {}
local next_cells = {}
local sim_timer = 0.0
local generation = 0
local paused = false

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Conway's Game of Life")
    for i = 0, 299 do
        if love.math.random() > 0.7 then
            table.insert(cells, 1.0)
        else
            table.insert(cells, 0.0)
        end
        table.insert(next_cells, 0.0)
    end
end

function love.keypressed(k)
    if k == "space" then
        paused = not paused
    end
end

function love.update(dt)
    if not paused then
        sim_timer = sim_timer + dt
        if sim_timer >= 0.2 then
            sim_timer = 0.0
            generation = generation + 1
            for row = 0, 14 do
                for col = 0, 19 do
                    local count = 0
                    if row > 0 and col > 0 and cells[(row - 1) * 20 + col - 1] == 1.0 then count = count + 1 end
                    if row > 0 and cells[(row - 1) * 20 + col] == 1.0 then count = count + 1 end
                    if row > 0 and col < 19 and cells[(row - 1) * 20 + col + 1] == 1.0 then count = count + 1 end
                    if col > 0 and cells[row * 20 + col - 1] == 1.0 then count = count + 1 end
                    if col < 19 and cells[row * 20 + col + 1] == 1.0 then count = count + 1 end
                    if row < 14 and col > 0 and cells[(row + 1) * 20 + col - 1] == 1.0 then count = count + 1 end
                    if row < 14 and cells[(row + 1) * 20 + col] == 1.0 then count = count + 1 end
                    if row < 14 and col < 19 and cells[(row + 1) * 20 + col + 1] == 1.0 then count = count + 1 end
                    local idx = row * 20 + col
                    if cells[idx] == 1.0 then
                        if count == 2 or count == 3 then
                            next_cells[idx] = 1.0
                        else
                            next_cells[idx] = 0.0
                        end
                    else
                        if count == 3 then
                            next_cells[idx] = 1.0
                        else
                            next_cells[idx] = 0.0
                        end
                    end
                end
            end
            for i = 0, 299 do
                cells[i] = next_cells[i]
            end
        end
    end
end

function love.draw()
    for row = 0, 14 do
        for col = 0, 19 do
            if cells[row * 20 + col] == 1.0 then
                love.graphics.rectangle("fill", col * 40.0, row * 40.0, 38.0, 38.0)
            end
        end
    end
    love.graphics.print("Gen: " .. generation, 10.0, 10.0)
    if paused then
        love.graphics.print("PAUSED", 350.0, 10.0)
    end
end

File: vibe-lang/benchmark/output/python-pygame/openai/cellular_automata.py
import pygame
import sys
import random

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Conway's Game of Life")
clock = pygame.time.Clock()

# Constants
GRID_WIDTH = 20
GRID_HEIGHT = 15
CELL_SIZE = 40

# Game state variables
cells = []
next_cells = []
sim_timer = 0.0
generation = 0
paused = False

def load():
    global cells, next_cells
    for i in range(0, GRID_WIDTH * GRID_HEIGHT):
        if random.random() > 0.7:
            cells.append(1.0)
        else:
            cells.append(0.0)
        next_cells.append(0.0)

def keypressed(k):
    global paused
    if k == pygame.K_SPACE:
        paused = not paused

def update(dt):
    global sim_timer, generation
    if not paused:
        sim_timer += dt
        if sim_timer >= 0.2:
            sim_timer = 0.0
            generation += 1
            for row in range(0, GRID_HEIGHT):
                for col in range(0, GRID_WIDTH):
                    count = 0
                    # Check 8 neighbors
                    if row > 0 and col > 0 and cells[(row - 1) * GRID_WIDTH + col - 1] == 1.0:
                        count += 1
                    if row > 0 and cells[(row - 1) * GRID_WIDTH + col] == 1.0:
                        count += 1
                    if row > 0 and col < GRID_WIDTH - 1 and cells[(row - 1) * GRID_WIDTH + col + 1] == 1.0:
                        count += 1
                    if col > 0 and cells[row * GRID_WIDTH + col - 1] == 1.0:
                        count += 1
                    if col < GRID_WIDTH - 1 and cells[row * GRID_WIDTH + col + 1] == 1.0:
                        count += 1
                    if row < GRID_HEIGHT - 1 and col > 0 and cells[(row + 1) * GRID_WIDTH + col - 1] == 1.0:
                        count += 1
                    if row < GRID_HEIGHT - 1 and cells[(row + 1) * GRID_WIDTH + col] == 1.0:
                        count += 1
                    if row < GRID_HEIGHT - 1 and col < GRID_WIDTH - 1 and cells[(row + 1) * GRID_WIDTH + col + 1] == 1.0:
                        count += 1

                    idx = row * GRID_WIDTH + col
                    if cells[idx] == 1.0:
                        if count == 2 or count == 3:
                            next_cells[idx] = 1.0
                        else:
                            next_cells[idx] = 0.0
                    else:
                        if count == 3:
                            next_cells[idx] = 1.0
                        else:
                            next_cells[idx] = 0.0

            # Copy next_cells to cells
            for i in range(0, GRID_WIDTH * GRID_HEIGHT):
                cells[i] = next_cells[i]

def draw():
    screen.fill((0, 0, 0))
    for row in range(0, GRID_HEIGHT):
        for col in range(0, GRID_WIDTH):
            if cells[row * GRID_WIDTH + col] == 1.0:
                pygame.draw.rect(screen, (255, 255, 255), (col * CELL_SIZE, row * CELL_SIZE, CELL_SIZE - 2, CELL_SIZE - 2))
    font = pygame.font.Font(None, 36)
    text = font.render("Gen: " + str(generation), True, (255, 255, 255))
    screen.blit(text, (10, 10))
    if paused:
        paused_text = font.render("PAUSED", True, (255, 255, 255))
        screen.blit(paused_text, (350, 10))

load()

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.KEYDOWN:
            keypressed(event.key)

    update(dt)
    draw()

    pygame.display.flip()

pygame.quit()
sys.exit()